motorutil/steer.go

```diff
@@ -0,0 +1,232 @@
+// Copyright ©2016 The ev3go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package motorutil
+
+import (
+	"fmt"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/ev3go/ev3dev"
+)
+
+// Steering implements a paired-motor steering unit similar to an EV-G steering block.
+type Steering struct {
+	// Left and Right are the left and right motors to be
+	// used by the steering module.
+	Left, Right *ev3dev.TachoMotor
+
+	// Timeout is the timeout for waiting for motors to
+	// return to a non-driving state.
+	//
+	// See ev3dev.Wait documentation for timeout behaviour.
+	Timeout time.Duration
+}
+
+// SteerCounts steers in the given turn for the given tacho counts and at the
+// specified speed. The valid range of turn is -100 (hard left) to +100 (hard right).
+// If counts is negative, the turn will be made in reverse. The sign of speed is
```

Comment:
>Ignoring the sign of speed is another thing that, based on user feedback, is non-intuitive.

This is merely following the underlying ev3dev `speed_sp` behaviour. If that changes, I'll update this.

Reply:
I was thinking of taking that approach, but the documentation becomes more complex in that then it would need to say the direction of travel is the sign of the product of counts and speed.

Reply:
The docs may be more complex, but users expect negative speeds to "just work" before ever consulting the documentation (at least in my experience).

Reply:
What is the expected behaviour of -speed and -position in conjunction? This is the problem for me.

Reply:
Just like multiplication in algerbra, two negative make a positive. Also, it works this way in EV3-G.

<img width="488" alt="image" src="https://user-images.githubusercontent.com/963645/30569837-c72c7550-9ca2-11e7-98ae-793e9010dd70.png">

does the exact same thing as 

<img width="489" alt="image" src="https://user-images.githubusercontent.com/963645/30569844-d5f9ddb6-9ca2-11e7-9152-b9723a4012d9.png">

Both drive forwards and then backwards

Reply:
OK, it's a deal. I'll make that change tonight.

Reply:
Done.